CedarJava/src/main/java/com/cedarpolicy/value/EntityUID.java

```diff
@@ -17,118 +17,104 @@
 package com.cedarpolicy.value;
 
 import java.util.Optional;
-import java.util.regex.Matcher;
-import java.util.regex.Pattern;
-import java.util.regex.PatternSyntaxException;
+import java.util.Objects;
+
+import com.cedarpolicy.serializer.JsonEUID;
 
 /**
  * Represents a Cedar Entity UID. An entity UID contains both the entity type and a unique
  * identifier for the entity formatted as <code>TYPE::"ID"</code>.
  */
-public class EntityUID extends Value {
-
-    private static class EUIDValidator {
```

Comment:
I think your code is obviously better, but should probably get Barbara to sign off on removing this.